File: Algorithms/Recursive_algorithms/sum_of_elements.c
// Write a recursive function to find the sum of elements in an array

#include <stdio.h>

int sumArray(int arr[], int n) {
    if (n == 0) {
        return 0;
    } else {
        return arr[n - 1] + sumArray(arr, n - 1);
    }
}

int main() {
    int arr[] = {2, 4, 6, 8, 10};
    int n = sizeof(arr) / sizeof(arr[0]);
    int result = sumArray(arr, n);
    printf("Sum of elements in the array is %d\n", result);
    return 0;
}

File: Algorithms/Recursive_algorithms/sum_of_elements.py
# Write a recursive function to find the sum of elements in an array

def sum_array(arr, n):
    if n == 0:
        return 0
    else:
        return arr[n - 1] + sum_array(arr, n - 1)

arr = [2, 4, 6, 8, 10]
n = len(arr)
result = sum_array(arr, n)
print("Sum of elements in the array is", result)